### PR TITLE
Improve 7.6 release notes

### PR DESCRIPTION
**Pull request by ljacomet**
* Rework dependency block release notes
* Rework preamble and section ordering in release notes and match release features

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Review by eskatos — APPROVED** [review]

**Review by hythloda — APPROVED** [review]
Only comment I have is it is not clear in the preamble why a strongly-typed dependencies block matters.

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -172,9 +147,72 @@ Using a `FileCollection` provides a `FileCollectionDependency`.
 This allows Java and Kotlin to properly configure all types of dependencies
 and improves IDE support for the Groovy DSL.
 
+In addition, the Kotlin DSL now supports named arguments for external dependencies:
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:150]: Should we change this section title to reflect this additional information?

Comment (ljacomet) [reply subprojects/docs/src/docs/release/notes.md:150]: I do not think that is necessary. But I am happy to take suggestions from @big-guy or @octylFractal

**Comment by ljacomet** [comment]
> Only comment I have is it is not clear in the preamble why a strongly-typed dependencies block matters.

@big-guy @octylFractal Any idea on clarifying that

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -370,39 +395,6 @@ can be passed from the command line as follows:
...
-        }
-    }
-}
-```
-
-For more information about using custom toolchain resolvers, see the [Toolchain Download Repositories documentation](userguide/toolchains.html#sub:download_repositories).
-For more information about writing custom toolchain resolvers, see the [Toolchain Resolver Plugins documentation](userguide/toolchain_plugins.html).
-
 <a name="ide"></a>
 ### IDE
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:399]:
```suggestion
### IDE Integration
```
Piotr's Suggestion from Google doc

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -196,22 +234,27 @@ accepting connections via network on Java 9 and above.
 On Java 9 and above, use the special host address value `*` to make the debugger server listen on all network interfaces.
 Otherwise, use the address of one of the machine's network interfaces.
 
+#### Revised dependencies generated by Maven conversions from `implementation` to `api`
+
+The `init` task now adds compile-time Maven dependencies to Gradle's `api` configuration
+when converting a Maven project. This sharply reduces the number of compilation errors.
+For more information about Maven conversions, see the [Build Init Plugin](userguide/build_init_plugin.html#sec:pom_maven_conversion).
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:241]:
```suggestion
For more information about Maven conversions, see the [Build Init Plugin](userguide/build_init_plugin.html#sec:pom_maven_conversion).

It is still recommended to use `implementation` where possible.
```

**Review by hythloda — COMMENTED** [review]
Added Piotr's first feedback

**Comment by ljacomet** [comment]
Included suggested changes. Will merge so that we are good to go for RC1. We can polish further before GA.

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
Something needs to be cherry-picked to `release`, on it.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -343,6 +349,28 @@ The following Attributes have disambiguation rules defined.
 For more information, see [Attribute Disambiguation Rules](userguide/variant_attributes.html#sec:abm_disambiguation_rules).
 
 
+
+<a name="configuration"></a>
+### Configuration Cache
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:354]: Piotr: I'd add an introductory paragraph about the configuration cache similarly to previous release notes.

File: subprojects/docs/src/docs/release/notes.md [hunk 5]
```diff
@@ -343,6 +349,28 @@ The following Attributes have disambiguation rules defined.
...
+
+#### Improved configuration cache failure recovery
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:356]: Piotr: It's not entirely clear to me after reading this what's the point of having config cache enabled / disabled. It seems the main point is automatic recovery from failures anyway.

File: subprojects/docs/src/docs/release/notes.md [hunk 6]
```diff
@@ -223,12 +268,6 @@ Gradle now places convention plugins under the `build-logic` directory instead o
 
 For more information about convention plugins, see [Convention Plugins](userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins).
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:269]: Piotr: Need to add why this is useful.

**Review by hythloda — COMMENTED** [review]
Piotr's feedback

File: subprojects/docs/src/docs/release/notes.md (hunk 6 above)
Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:269]: @big-guy can you add?

File: subprojects/docs/src/docs/release/notes.md [hunk 7]
```diff
@@ -343,6 +353,28 @@ The following Attributes have disambiguation rules defined.
 For more information, see [Attribute Disambiguation Rules](userguide/variant_attributes.html#sec:abm_disambiguation_rules).
 
 
+
+<a name="configuration"></a>
+### Configuration Cache
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:358]:
```suggestion
### Configuration Cache

The [configuration cache](https://docs.gradle.org/7.5/userguide/configuration_cache.html) improves build time by caching the result of the configuration phase and reusing this for subsequent builds.
```

File: subprojects/docs/src/docs/release/notes.md (hunk 5 above)
Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:362]: @mlopatkin can you address this?

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (octylFractal) [reply subprojects/docs/src/docs/release/notes.md:150]: I agree, I think it works as-is.

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/docs/release/notes.md [hunk 8]
```diff
@@ -99,33 +96,11 @@ vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
...
-    }
-}
-```
-
 <a name="strongly-typed-dependencies"></a>
 #### Introduced strongly-typed `dependencies` block for JVM test suites
 
 The [JVM test suite](userguide/jvm_test_suite_plugin.html) `dependencies` block
-now uses a [strongly-typed API](dsl/org.gradle.api.plugins.jvm.JvmComponentDependencies.html).
+now uses a [strongly-typed API](dsl/org.gradle.api.plugins.jvm.JvmComponentDependencies.html). This makes the build logic cleaner and improving assistance in the IDEs, especially with Kotlin DSL.
```
[the remark this answers is not in the document]
Comment (ljacomet) [reply subprojects/docs/src/docs/release/notes.md:103]: I also changed improving -> improves

File: subprojects/docs/src/docs/release/notes.md [hunk 9]
```diff
@@ -172,9 +147,72 @@ Using a `FileCollection` provides a `FileCollectionDependency`.
 This allows Java and Kotlin to properly configure all types of dependencies
 and improves IDE support for the Groovy DSL.
 
+In addition, the Kotlin DSL now supports named arguments for external dependencies in this block:
+
+```kotlin
+testing {
+    suites {
+        val test by getting(JvmTestSuite::class) {
```
Comment (bamboo) [inline subprojects/docs/src/docs/release/notes.md:155]: We shouldn't be promoting the `by getting` pattern in the release notes for two reasons:
1. `getting` is the eager accessor
2. the `val ... by ...` idiom is considered unfriendly to novices
The novice friendly configuration idiom is:
```kotlin
    suites {
        "test"(JvmTestSuite::class) { ...
```

File: subprojects/docs/src/docs/release/notes.md (hunk 5 above)
Comment (bamboo) [reply subprojects/docs/src/docs/release/notes.md:362]: I see an explanation at the beginning of the _Configuration Cache_ section now and I also rephrased the recovery entry for clarity.

File: subprojects/docs/src/docs/release/notes.md (hunk 9 above)
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:155]:
```suggestion
        "test"(JvmTestSuite::class) { ...
```

Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:155]: this correct @bamboo

File: subprojects/docs/src/docs/release/notes.md (hunk 6 above)
Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md:269]: Short: Convention plugins is Gradle’s recommended way of organizing build logic.

Long: You can compose custom build logic by applying and configuring both core and external plugins and create custom convention plugins that define new project types and configure conventions specific to your project or organization.

Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:265]: Thanks! added

**Comment by hythloda** [comment]
This PR also incorporated Piotr's Feedback from the google doc.  All points have been addressed and I think this can be safely merged.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**